File: src/prototype/Circle.java
package prototype;

public class Circle implements ShapePrototype{
    private String color;
    private int radius;

    public Circle(int radius) {
        this.radius = radius;
    }

    @Override
    public ShapePrototype clone() {
        Circle clone = new Circle(this.radius);
        clone.setColor(this.color);
        return clone;
    }

    @Override
    public void draw() {
        System.out.println("Dibujando círculo - Radio: " + radius
                + ", Color: " + color);
    }

    @Override
    public void setColor(String color) {
        this.color = color;
    }

    @Override
    public String getColor() {
        return color;
    }
}

File: Python/python/Prototype/Circle.py
import copy
from .ShapePrototype import ShapePrototype

# Implementación concreta del prototipo: Círculo
class Circle(ShapePrototype):
    def __init__(self, radius: int):
        self._radius = radius
        self._color = None

    def clone(self):
        return copy.deepcopy(self)

    def draw(self):
        print(f"Dibujando círculo - Radio: {self._radius}, Color: {self._color}")

    def set_color(self, color: str):
        self._color = color

    @property
    def color(self):
        return self._color